### PR TITLE
[DOCS] how to use docker images

### PR DESCRIPTION
**Pull request by Aylr**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [ENHANCEMENT], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- docs for docker usage
-
-

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

Thank you for submitting!

### CODE REVIEW

**Review by jcampbell — CHANGES_REQUESTED** [review]
Capturing several changes that are actually more about the build process here:

1. We do need to provide a `latest` tag. People may not use it, and we may not *recommend* it, but it's definitely a useful convenience for getting started.

2. We may need to revisit the hosting path (separate thread already started previously).

3. Rather than a huge fat base image, I would like to see a base ge install that only requires the required dependencies, with additional images that add additional features (like spark or sqlalchemy)

File: docs/guides/how_to_guides/miscellaneous/how_to_use_official_docker_images.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        docker run \
+        -v /full/path/to/your/project/great_expectations:/usr/app/great_expectations \
+        greatexpectations/great_expectations:python-3.7-buster-ge-0.12.0 \
+        datasource list
+
+
+Additional notes
+----------------
+
+If you need to run interactive ``great_expectations`` commands, this is best done from inside the container by running Docker in interactive mode and changing the entrypoint as follows:
```
Comment (jcampbell) [inline docs/guides/how_to_guides/miscellaneous/how_to_use_official_docker_images.rst]: I actually liked the design of a consistent entrypoint (ge); what is the advantage of recommending changing the entrypoint?

**Comment by ddaltonrtr** [comment]
> Capturing several changes that are actually more about the build process here:
> 
> 1. We do need to provide a `latest` tag. People may not use it, and we may not _recommend_ it, but it's definitely a useful convenience for getting started.
> 2. We may need to revisit the hosting path (separate thread already started previously).
> 3. Rather than a huge fat base image, I would like to see a base ge install that only requires the required dependencies, with additional images that add additional features (like spark or sqlalchemy)

I have a couple comments to build upon this. I strongly agree with point number 3. It would be great if we could avoid having the dependencies installed for things that we do not use at this time, such as spark, gcp, redshift, etc. One longer-term suggestion may be to follow the setup Fishtown Analytics does with dbt. You can install dbt as a package and get everything, or you can install dbt-snowflake and just get dbt & snowflake dependencies (dbt-redshift would give dbt & redshift dependencies only). It's a convenient way to parse down unnecessary dependencies.

One comment I would suggest is having a non-root user used in the end. For us, we we don't run containers as root users and it is always helpful if the image has a preferred/default user to use (For example, rshiny server has the shiny user).

I am copying the dockerfile we have for GE here:
```
FROM python:3.7.6-slim AS reqs

ENV DIRPATH /great_expectations

COPY requirements-ge.txt .

ARG USER_ID

ARG GROUP_ID

# Install requirements
RUN addgroup --gid $GROUP_ID ge_user \
	&& useradd -l --uid $USER_ID --gid $GROUP_ID ge_user -m -d $DIRPATH \
    && pip --no-cache-dir \
        install \
        -r requirements-ge.txt

FROM reqs

USER ge_user

WORKDIR $DIRPATH

# Shell script to be executed as the entrypoint
COPY ge_entrypoint_prod.sh .

EXPOSE 8888

ENTRYPOINT ["./ge_entrypoint_prod.sh"] 
```

(I don't expect the entrypoint to be the same). I will say that one nice thing about this image is that we can either run `great_expectations` commands or `python` commands. We use this image to both create/edit expectations locally and run expectations against datasets in prod (Using the `python` command with a script built upon a checkpoint).

In our requirements-ge.txt file we have great-expectations and also aws-cli since we sync the docs to S3 and also use this image to do so.

File: docs/guides/how_to_guides/miscellaneous/how_to_use_official_docker_images.rst (hunk 1 above)
Comment (Aylr) [reply docs/guides/how_to_guides/miscellaneous/how_to_use_official_docker_images.rst]: Interactive commands require an actual shell. It's not great, but my hope is that this image is mostly used for deployments really.

Comment (jcampbell) [reply docs/guides/how_to_guides/miscellaneous/how_to_use_official_docker_images.rst]: I might just be getting confused at wording, but interactive commands work fine with the current entrypoint and not dropping into a shell as long as the appropriate flags (`-it`) are included.

Comment (Aylr) [reply docs/guides/how_to_guides/miscellaneous/how_to_use_official_docker_images.rst]: Ah! Now I see what you mean! I have used `-it` + an `bin/bash` for so long I didn't realize you could just use the simpler way! Thank you.

**Comment by Aylr** [comment]
> 1. We do need to provide a `latest` tag. People may not use it, and we may not _recommend_ it, but it's definitely a useful convenience for getting started.
> 2. We may need to revisit the hosting path (separate thread already started previously).
> 3. Rather than a huge fat base image, I would like to see a base ge install that only requires the required dependencies, with additional images that add additional features (like spark or sqlalchemy)

@jcampbell I wish that we had raised this feedback in the initial Docker image PR as these are substantial amounts of work. I move we proceed forward as is and iteratively implement these good ideas in the future.

**Comment by jcampbell** [comment]
There is no marginal work in #1 or #2.

#3 is just about ensuring that the public, recommended image does not have a lot of bloat, i.e. that our recommended practices comply with generally accepted best practices for using docker.

**Review by eugmandel — APPROVED** [review]

**Pull request merged**